[OKTA-802677] Updates to SCIM 2.0 with entitlements (#5080)
okta/okta-developer-docs: packages/@okta/vuepress-site/docs/guides/scim-with-entitlements/main/index.md

#### What you need

> * [Okta Developer Edition organization](https://developer.okta.com/signup)

Reviewer: ```suggestion
* [Okta Developer Edition org](https://developer.okta.com/signup)
```

* Some basic development experience with the SCIM [core schema](https://datatracker.ietf.org/doc/html/rfc7643) and [protocol](https://datatracker.ietf.org/doc/html/rfc7644)

> * [Okta Identity Governance](https://help.okta.com/okta_help.htm?type=oie&id=ext-iga) in order to use [entitlements](https://help.okta.com/okta_help.htm?type=oie&id=ext-entitlement-mgt)

Reviewer: ```suggestion
* [Okta Identity Governance](https://help.okta.com/okta_help.htm?type=oie&id=ext-iga) to use [entitlements](https://help.okta.com/okta_help.htm?type=oie&id=ext-entitlement-mgt)
```

You don't need a `/Schemas` endpoint for apps that don't use custom schema extensions. Okta handles the base definitions for the following resources and objects:

* [`Entitlements`](#entitlements-schema)
* [`Roles`](#roles-schema)
* `User:Core`
* `User:Enterprise`

#### Entitlements schema

The `urn:okta:scim:schemas:core:1.0:Entitlement` schema contains the following information:

|Parameter | Type | Description | Notes |
|---|---|---|---|
| `id` | String | The ID of the entitlement | Required |

> | `displayName` | String | The name displayed in the UI for the entitlement | Required|

Reviewer: ```suggestion
| `displayName` | String | The name that appears in the Admin Console for the entitlement | Required|
```

| `type` | String | Corresponds with the `ResourceType` name field for the entitlement, such as "License" or "Permission set" | Required |

> | `description` | String | A human-readable description of the entitlement. This appears in the Governance tab. | Optional, max length 1000 characters |

Reviewer: ```suggestion
| `description` | String | A human-readable description of the entitlement. This appears in the **Governance** tab. | Optional, max length 1000 characters |
```

See [Custom entitlement with extensions](#custom-entitlement-with-extensions) for an example schema.

#### Roles schema

The `urn:okta:scim:schemas:core:1.0:Entitlement` schema contains the following information:

|Parameter | Type | Description | Notes |
|---|---|---|---|
| `id` | String | The ID of the role | Required |

> | `displayName` | String | The name displayed in the UI for the role | Required|

Reviewer: ```suggestion
| `displayName` | String | The name that appears in the Admin Console for the role | Required|
```
